rmw_fastrtps_cpp/src/rmw_node.cpp

```diff
@@ -168,14 +168,25 @@ get_security_file_paths(
 
   const char * file_prefix = "file://";
 
-  std::string tmpstr;
+  char * file_path = nullptr;
   for (size_t i = 0; i < num_files; i++) {
-    tmpstr = std::string(rcutils_join_path(node_secure_root, file_names[i]));
-    if (!rcutils_is_readable(tmpstr.c_str())) {
+    file_path = rcutils_join_path(node_secure_root, file_names[i]);
+    if (!file_path) {
+      RMW_SET_ERROR_MSG("Failed to allocate memory for security file path");
       return false;
     }
-    security_files_paths[i] = std::string(file_prefix + tmpstr);
+
+    if (rcutils_is_readable(file_path)) {
+      security_files_paths[i] = std::string(file_prefix + std::string(file_path));
```

Comment:
While this works, it is faster to make `file_prefix` above a std::string (like `std::string file_prefix("file://");`, and then here, do:

```
security_file_paths[i] = file_prefix + std::string(file_path);
```

Once we have that, and reduce the scope of the `file_path` like @wjwwood suggested, I think this will be ready to go.